wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta4.0/1__add_devfile.sql

```diff
@@ -0,0 +1,250 @@
+    tag                  VARCHAR(255),
+    commit_id            VARCHAR(255),
+    devfile_id           BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_project ADD CONSTRAINT fk_devfile_projects_id FOREIGN KEY (devfile_id) REFERENCES devfile (id);
+CREATE UNIQUE INDEX index_devfile_project_name ON devfile_project (devfile_id, name);
+
+-----------------------------------------------------------------------------------------------------------------------------------
+
+-- devfile command
+CREATE TABLE devfile_command (
+    id                  BIGINT       NOT NULL,
+    name                VARCHAR(255) NOT NULL,
+    devfile_id          BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_command ADD CONSTRAINT fk_devfile_command_id FOREIGN KEY (devfile_id) REFERENCES devfile (id);
+CREATE UNIQUE INDEX index_devfile_command_name ON devfile_command (devfile_id, name);
+
+-- devfile command attributes
+CREATE TABLE devfile_command_attributes (
+    devfile_command_id     BIGINT,
+    value                  TEXT,
+    name                   VARCHAR(255)
+);
+-- constraints & indexes
+ALTER TABLE devfile_command_attributes ADD CONSTRAINT fk_devfile_command_attributes_command_id FOREIGN KEY (devfile_command_id) REFERENCES devfile_command (id);
+CREATE UNIQUE INDEX index_devfile_command_attributes_name ON devfile_command_attributes (devfile_command_id, name);
+
+-- devfile command action
+CREATE TABLE devfile_action (
+    id                      BIGINT       NOT NULL,
+    type                    VARCHAR(255) NOT NULL,
+    component               VARCHAR(255) NOT NULL,
+    command                 TEXT NOT NULL,
+    workdir                 TEXT,
+    devfile_command_id      BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_action ADD CONSTRAINT fk_devfile_actions_id FOREIGN KEY (devfile_command_id) REFERENCES devfile_command (id);
+CREATE INDEX index_action_command_id ON devfile_action (devfile_command_id);
+
+
+-----------------------------------------------------------------------------------------------------------------------------------
+
+-- devfile component
+CREATE TABLE devfile_component (
+    id                     BIGINT       NOT NULL,
+    name                   VARCHAR(255) NOT NULL,
+    type                   VARCHAR(255) NOT NULL,
+    component_id           TEXT,
+    reference              TEXT,
+    reference_content      TEXT,
+    image                  TEXT,
+    memory_limit           VARCHAR(255),
+    mount_sources          BOOLEAN,
+    devfile_id             BIGINT,
+
+    PRIMARY KEY (id)
+);
+
+-- constraints & indexes
+ALTER TABLE devfile_component ADD CONSTRAINT fk_devfile_component_id FOREIGN KEY (devfile_id) REFERENCES devfile (id);
+CREATE UNIQUE INDEX index_devfile_component_name ON devfile_component (devfile_id, name);
+
+
+-- component command
+CREATE TABLE component_command (
+    devfile_component_id    BIGINT,
```

Comment:
can not it be just `component_id `?

Reply:
To be more consistent with parent entity table name as requested by @skabashnyuk https://github.com/eclipse/che/pull/13111#discussion_r274774659

Reply:
@skabashnyuk Can you comment? Why do you request it? Do you think that it is better to have a foreign key field as `referenced table name` + `_id`?
Do you think that we should prefix all devfile related tables with `devfile_`? I mean we have table named as `devfile_component`, and `component_command` or `component_arg`. Should not it be changed  to `devfile_component_command`, `devfile_component_arg`?

Reply:
We should have `devfile_` prefix for at least parent entities due to condlicts with ols WS API entitien with same name like `command` etc

Reply:
We did not have `component` table before, so nothing prevents from having `component` table.
I'm OK with `devfile_component` table name, and the question here is if we should use `referenced table name` + `_id` for foreign key field, then should not we use `parentTableName` + `_` + `tableName` for child tables? If yes, then `component_command` should be `devfile_component_command`

Reply:
Let's put ```devfile_``` prefix to all tables associated with defvile functionality to avoid collisions with other table names.